src/libcore/macros.rs

```diff
@@ -542,7 +542,9 @@ macro_rules! unreachable {
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
 macro_rules! unimplemented {
-    () => (panic!("not yet implemented"))
+    () => (panic!("not yet implemented"));
+    ($msg:expr) => (unimplemented!("{}", $msg));
```

Comment:
I know this matches `unreachable` but does this macro really need to accept anything that implements `Display` as a single argument? Is it that useful to be able to use `unimplemented!(x)` rather than `unimplemented!("{}", x)` given that it means `unimplemented!("Conn::poll; state={:?}")` would compile when it arguably shouldn't?